### PR TITLE
Add show methods for remotes and tags

### PR DESCRIPTION
**Pull request by kshyatt**
Re: #19839

Oh, I added a `name` function for `remote` since we didn't have that and it seems useful to know.

### CODE REVIEW

File: base/libgit2/remote.jl [hunk 1]
```diff
@@ -88,3 +94,5 @@ function push{T<:AbstractString}(rmt::GitRemote, refspecs::Vector{T};
         !no_refs && close(sa)
     end
 end
+
+Base.show(io::IO, rmt::GitRemote) = print(io, "Remote name: ", name(rmt), " url: ", url(rmt))
```
Comment (simonbyrne) [inline base/libgit2/remote.jl]: I would suggest printing the name of the type.

**Review by simonbyrne — CHANGES_REQUESTED** [review]

**Review by simonbyrne — APPROVED** [review]

File: base/libgit2/tag.jl [hunk 2]
```diff
@@ -41,3 +41,5 @@ function target(tag::GitTag)
     oid_ptr == C_NULL && throw(Error.GitError(Error.ERROR))
     return GitHash(oid_ptr)
 end
+
+Base.show(io::IO, tag::GitTag) = print(io, "GitTag:\nTag name: $(name(tag)) target: $(target(tag))")
```
Comment (tkelman) [inline base/libgit2/tag.jl]: kinda inconsistent to have one of these use interpolation and the other use multi-arg print

File: base/libgit2/remote.jl [hunk 3]
```diff
@@ -35,7 +35,13 @@ end
 function url(rmt::GitRemote)
     url_ptr = ccall((:git_remote_url, :libgit2), Cstring, (Ptr{Void}, ), rmt.ptr)
     url_ptr == C_NULL && return ""
-    return  unsafe_string(url_ptr)
+    return unsafe_string(url_ptr)
+end
+
+function name(rmt::GitRemote)
+    name_ptr = ccall((:git_remote_name, :libgit2), Cstring, (Ptr{Void}, ), rmt.ptr)
+    name_ptr == C_NULL && return ""
```
Comment (tkelman) [inline base/libgit2/remote.jl]: is it valid to have empty-name or empty-url remotes?

Comment (kshyatt) [inline base/libgit2/remote.jl]: You can have `Anonymous` remotes, can't you?

Comment (kshyatt) [inline base/libgit2/remote.jl]: Like [this](https://libgit2.github.com/libgit2/#HEAD/group/remote/git_remote_create_anonymous).

Comment (tkelman) [inline base/libgit2/remote.jl]: Is that the only thing that can result in getting a null pointer back from this function?

**Pull request merged**